src/AuthModule/components/ChangPass/ChangPass.tsx

```diff
@@ -0,0 +1,195 @@
+
+const ChangPass: React.FC<ChangPassProps> = ({ handleClose }) => {
+  const { showSuccessToast, showErrorToast } = useToast();
+  const [spinner, setSpinner] = useState<boolean>(false);
+  const [showPassword, setShowPassword] = useState<boolean>(false);
+  const [showNewPassword, setShowNewPassword] = useState<boolean>(false);
+
+  const [showPasswordConfirm, setShowPasswordConfirm] =
+    useState<boolean>(false);
+
+  const togglePasswordVisibility = () => {
+    setShowPassword((prevState) => !prevState);
+  };
+
+  const toggleNewPassword = () => {
+    setShowNewPassword((prevState) => !prevState);
+  };
+
+  const togglePasswordVisibilityconfirm = () => {
+    setShowPasswordConfirm((prevState) => !prevState);
+  };
+
+  const {
+    register,
+    handleSubmit,
+    watch,
+    formState: { errors },
+  } = useForm<FormValues>();
+  // Custom validation function to check if passwords match
+  const validatePasswordMatch = (value: unknown) => {
+    const password = watch("newPassword"); // Get the value of the "password" field
+    return value === password || "Passwords do not match"; // Return error message if passwords don't match
+  };
+
+  const onSubmit: SubmitHandler<FormValues> = async (data) => {
+    setSpinner(true);
+    console.log(data);
+    const token = localStorage.getItem("adminToken");
+    if (!token) {
+      throw new Error("User is not authenticated");
+    }
+    try {
+      const response = await axios.put(
+        "https://upskilling-egypt.com:3003/api/v1/Users/ChangePassword",
+        data,
+        {
+          headers: {
+            Authorization: token,
+          },
+        }
+      );
+      console.log(response)
+      handleClose();
+      showSuccessToast("Password has been updated successfully");
+    } catch (error) {
+      if (axios.isAxiosError(error) && error.response) {
+        showErrorToast(error.response.data.message);
+      }
+    } finally {
+      setSpinner(false);
+    }
+  };
+
+  return (
+    <>
+      <div className=" row justify-content-center align-items-center">
+        <div className=" col-md-12">
+          <div className="  rounded-3 px-5 py-4">
+            <small className=" text-white">welcome to PMS</small>
+
+            <h4 className={`${styleChangPass.changeWord} mb-5 `}>
+              {" "}
+              Change Password
+              <div className="color "></div>
+            </h4>
+
+            <form onSubmit={handleSubmit(onSubmit)}>
+              <div className="form-group conInput  mb-3">
+                <label className=" color-text" htmlFor="old ">
+                  Old Password
+                </label>
+                <input
+                  className={`${styleChangPass.styleInput} form-control mt-2 `}
+                  type={showPassword ? "text" : "password"}
+                  placeholder="Enter your Old Password  "
+                  {...register("oldPassword", {
+                    required: "old Password is required ",
```

Comment:
We might add a pattern validation here, to ensure that the network request is not sent to the backend unless all the data match the validation. This will significantly reduce the total number of unnecessary requests sent to the backend.